12/umbraco-cms/reference/content-delivery-api/media-delivery-api.md

```diff
@@ -0,0 +1,265 @@
+{% swagger-description %}
+Returns a single item.
+{% endswagger-description %}
+
+{% swagger-parameter in="path" name="path" type="String" required="true" %}
+Path of the media item. The path is composed by the names of any ancestor folders and the name of the media item itself, separated by `/`.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="header" name="Api-Key" type="String" required="false" %}
+Access token
+{% endswagger-parameter %}
+
+{% swagger-parameter in="query" name="expand" type="String" required="false" %}
+Which properties to expand in the response
+{% endswagger-parameter %}
+
+{% swagger-response status="200: OK" description="Media item" %}
+
+{% endswagger-response %}
+
+{% swagger-response status="401: Unauthorized" description="Missing permissions after protection is set up" %}
+
+{% endswagger-response %}
+
+{% swagger-response status="404: Not Found" description="Media item not found" %}
+
+{% endswagger-response %}
+{% endswagger %}
+
+{% swagger method="get" path="/media" baseUrl="/umbraco/delivery/api/v1" summary="Gets media item(s) from a query" %}
+{% swagger-description %}
+Returns single or multiple items.
+{% endswagger-description %}
+{% swagger-parameter in="query" name="fetch" type="String" required="true" %}
+Structural query string option (e.g. `ancestors`, `children`, `descendants`).
+
+**Please note:** The default API implementation only supports `children`.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="query" name="filter" type="String Array" required="false" %}
+Filtering query string options (e.g. `mediaType`, `name`)
+{% endswagger-parameter %}
+
+{% swagger-parameter in="query" name="sort" type="String Array" required="false" %}
+Sorting query string options (e.g. `createDate`, `name`, `sortOrder`, `updateDate`)
+{% endswagger-parameter %}
+
+{% swagger-parameter in="query" name="skip" type="Integer" required="false" %}
+Amount of items to skip
+{% endswagger-parameter %}
+
+{% swagger-parameter in="query" name="take" type="Integer" required="false" %}
+Amount of items to take
+{% endswagger-parameter %}
+
+{% swagger-parameter in="header" name="Api-Key" type="String" required="false" %}
+Access token
+{% endswagger-parameter %}
+
+{% swagger-parameter in="query" name="expand" type="String" required="false" %}
+Which properties to expand in the response
+{% endswagger-parameter %}
+
+{% swagger-response status="200: OK" description="Paginated list of media items" %}
+
+{% endswagger-response %}
+
+{% swagger-response status="400: Bad Request" description="Invalid request" %}
+
+{% endswagger-response %}
+
+{% endswagger %}
+
+{% swagger method="get" path="/media/item" baseUrl="/umbraco/delivery/api/v1" summary="Gets media item(s) by id" %}
+{% swagger-description %}
+Returns single or multiple items by id.
+{% endswagger-description %}
+
+{% swagger-parameter in="query" name="id" type="String Array" required="true" %}
```

Comment:
```suggestion
{% swagger-parameter in="query" name="id" type="String Array" required="false" %}
```

Reply:
The id array is absolutely (functionally) required, otherwise we have nothing to go on in terms of looking up items 😄